test/e2e/engine/template.go

```diff
@@ -150,6 +150,9 @@ func Build(cfg *config.Config, masterSubnetID string, agentSubnetID string, isVM
 	}
 
 	if config.EnableKMSEncryption && config.ClientObjectID != "" {
+		if prop.OrchestratorProfile.KubernetesConfig == nil {
```

Comment:
this is a nil panic vector in e2e runner if no `kubernetesConfig` object is present in input api model